### PR TITLE
Refactored hook handlers in ejabberd_ctl

### PR DESCRIPTION
**Pull request by pawlooss1**
This PR changes all hook handlers in `ejabberd_ctl` to `gen_hook` format.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/155388) / small_tests / 197c015d586f2290b8d8171657ce0350fbadcb3a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3887/155388/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3887/155388/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-12-01_13.05.56/lib.mongooseim.logs/run.2022-12-01_13.05.57/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/155813) / small_tests / 0983cb748ee3b603f5b5e9f70a1b36a8ffd097d3
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3887/155813/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3887/155813/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-12-02_11.48.44/lib.mongooseim.logs/run.2022-12-02_11.48.44/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/155872) / small_tests / c86979522928fcab7a2c46af25892e472178ebd2
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3887/155872/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3887/155872/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-12-02_12.47.57/lib.mongooseim.logs/run.2022-12-02_12.47.57/suite.log.html)

File: src/vcard/mod_vcard.erl [hunk 1]
```diff
@@ -390,7 +390,7 @@ process_local_iq(Acc, _From, _To, IQ = #iq{type = get}, _Extra) ->
                     From :: jid:jid(),
                     To :: jid:jid(),
                     IQ :: jlib:iq(),
-                    Extra :: gen_hook:extra()) ->
+                    Extra :: map()) ->
```
Comment (NelsonVides) [inline src/vcard/mod_vcard.erl:393]: Good catch, this one is not a hook 👌🏽

File: src/mod_roster.erl [hunk 2]
```diff
@@ -334,7 +334,16 @@ create_sub_el(Items, Version) ->
...
 get_user_roster(Acc, #{jid := #jid{luser = LUser, lserver = LServer}}, #{host_type := HostType}) ->
-    NewAcc = case mongoose_acc:get(roster, show_full_roster, false, Acc) of
+    NewAcc = get_user_roster(Acc, LUser, LServer, HostType),
+    {ok, NewAcc}.
+
+-spec get_user_roster(mongoose_acc:t(),
+                      jid:luser(),
+                      jid:lserver(),
+                      mongooseim:host_type()
+                    ) -> mongoose_acc:t().
```
Comment (NelsonVides) [inline src/mod_roster.erl:344]: I wouldn't separate that last parenthesis, if breaking the line I'd do so after the `->` 🤔

**Review by NelsonVides — COMMENTED** [review]
Minor comment on style, otherwise good cleanup 😄

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/156027) / small_tests / 389cb78aae0a092b0604c4061c0af46b0137467a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3887/156027/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3887/156027/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-12-05_09.09.42/lib.mongooseim.logs/run.2022-12-05_09.09.43/suite.log.html)

File: src/mod_roster.erl [hunk 3]
```diff
@@ -334,7 +334,15 @@ create_sub_el(Items, Version) ->
...
     Extra :: gen_hook:extra().
 get_user_roster(Acc, #{jid := #jid{luser = LUser, lserver = LServer}}, #{host_type := HostType}) ->
-    NewAcc = case mongoose_acc:get(roster, show_full_roster, false, Acc) of
+    NewAcc = get_user_roster(Acc, LUser, LServer, HostType),
+    {ok, NewAcc}.
+
+-spec get_user_roster(mongoose_acc:t(),
+                      jid:luser(),
+                      jid:lserver(),
+                      mongooseim:host_type()) ->mongoose_acc:t().
```
Comment (NelsonVides) [inline src/mod_roster.erl:343]:
```suggestion
                      mongooseim:host_type()) -> mongoose_acc:t().
```
😅

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/156082) / small_tests / 4deb53fa5498b9ea6e07ca7866cc3660ac478182
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3887/156082/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3887/156082/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-12-05_09.48.08/lib.mongooseim.logs/run.2022-12-05_09.48.08/suite.log.html)

**Pull request merged**